src/acl.c

```diff
@@ -2278,15 +2399,18 @@ void addACLLogEntry(client *c, int reason, int context, int argpos, sds username
  * ==========================================================================*/
 
 /* Add the formatted response from a single selector to the ACL GETUSER
- * response. This function returns the number of fields added. */
-int aclAddReplySelectorDescription(client *c, aclSelector *s) {
+ * response. This function returns the number of fields added. 
+ * 
+ * Setting verbose to 1 means that the full qualifier for key and channel
+ * permissions are shown.
+ */
+int aclAddReplySelectorDescription(client *c, aclSelector *s, int verbose) {
```

Comment:
Although this may be an esoteric case, it just becomes difficult to differentiate someone having all permissions on "%R~*" versus someone who has read permissions on "*". I don't know if that's a likely outcome, but it seemed worth guarding against. The alternative I was considering was just omitting all of the keys that weren't all permissions from the root permissions.

Reply:
sorry.. i didn't understand your response.
1. isn't `%R~*` and `read permissions on "*"` the same?
2. can people use the `%R~*` notation on the root selector?
3. are you saying that in the root segment, you rather fold `%RW~` into just `~` (for backwards compatibility), and the non-root selectors you rather be explicit (i.e. user who provided `~` will get back `%RW~`)?